test/FunctionalTests/Microsoft.OData.Client.Tests/ALinq/DataServiceContextMethodMockSampleTest.cs

```diff
@@ -0,0 +1,33 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="DataServiceContextMethodMockSampleTest.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using Moq;
+using Xunit;
+
+namespace Microsoft.OData.Client.Tests.ALinq
+{
+    /// <summary>
+    /// How to mock the DataServiceContext class methods using moq framework example.
+    /// </summary>
+
+    public class DataServiceContextMethodMockSampleTest

```

Comment:
>DataServiceContextMethodMockSampleTest [](start = 17, length = 38)

I don't think the test class is meaningful.

What we are testing is that we can test some functionalities using mock, not test mock itself.

For example, we can test some method is actually called when we call a certain method. And we don't care about that called method returns, we can use mock to mock that called method.